### ISSUE TITLE
Uploading edits in Wikibase preview

### ISSUE THREAD

**Issue by cooperzoe** [OpenRefine/OpenRefine#6305]
When I went to upload edits in the Wikibase preview, I expected to see a more prominent button telling me to **Upload Edits*making the next step in the process quite clear, instead of going through the extension menu.

<img width="1045" alt="Screenshot 2024-01-15 at 12 23 30 AM" src="https://github.com/OpenRefine/OpenRefine/assets/52141007/9262bb0f-ef4c-4acd-81b9-931ace6e08f1">

**Comment by t8210103** [comment]
Hello, i would like to be assigned to this issue.
Since it is an important modification in the UX, i am open to any proposed solutions.
The solution i will propose will be a few options/opinions on how it could possibly be changed, in order to make the process a bit more clear.
After a discussion on an appropriate change, maybe i could create a second PR where the final solution is being applied.

**Comment by wetneb** [comment]
Sounds great, thanks for volunteering!

**Comment by t8210103** [comment]
**Proposed Solutions:**
- The following concept design is simple but offers more direction to the Wikibase options.
   ![Screenshot 2024-05-18 190421](https://github.com/OpenRefine/OpenRefine/assets/115114218/011a0cab-b420-40f2-8ca4-960a737707e9)

- The following concept design enables the user to directly **see** and **select** the option to upload the edits, keeping the original "Extension" reference to the Wikibase.
   ![wikibase-first](https://github.com/OpenRefine/OpenRefine/assets/115114218/60fe03a4-23f5-4288-862c-3645b2d66395)

- The following concept design is a variation of the previous concept.
![wikibase-last](https://github.com/OpenRefine/OpenRefine/assets/115114218/6a275425-0b79-4c2b-85de-e4f523b5eb06)

- Finally, as another concept design, i have combined the two previous design logics offering both the advanced direction to the Wikibase options and the quick accessibility to the upload option. 
![combination](https://github.com/OpenRefine/OpenRefine/assets/115114218/76b9f851-5154-40e2-af05-f3be4bf094f4)


All the previous suggested changes are based on existing contents of translation.json files, so no extra translation content is required for them to be applied.

I am open to any comments and concerns according the introduced designs and i am also willing to attempt the implementation of whatever is preferred.

**Comment by wetneb** [comment]
Thanks for the mock-ups!

I would be reluctant to integrate this button in the grid view, because it's a view that is used very often so space there is very valuable. For people who don't use the Wikibase integration at all, that's a relatively big button that they will always see and have no use for.

I would rather add it inside the Preview tab (that becomes visible after the user has created a Wikibase schema), as suggested by the screenshot in @cooperzoe's issue description. She might be able to expand on the places where she'd expect such a button to be available. It could potentially also go in the schema tab I guess, but that one is already quite full perhaps? In terms of user workflow, I think it makes sense to want to upload edits after having previewed them, so I like the idea of adding it to the Preview tab.

**Comment by t8210103** [comment]
Yes of course! 

I am going to attempt placing the button only in the Preview tab that becomes visible after the creation of a Wikibase schema.
 
For the Preview tab, should the "Upload edits to Wikibase" be the only button available or should i make it like the second or third mock-up where both the button "Upload edits to Wikibase" and the dropdown "Wikibase" menu are available.

**Comment by wetneb** [comment]
I would keep the "Wikibase" menu visible at all times even if an upload button is available elsewhere, since it gives access to other features as well.

**Comment by t8210103** [comment]
Unfortunately i couldn't find a way to apply the change only in the preview tab.

The reason behind the tasks difficulty, is that the extension bar is created once for all the tabs, so changing only the extension bar of the preview tab (without affecting the others) would require a structural change to the creation and manipulation of the extension bar.

Since i am quite new to open source development, i couldn't find myself completing this task.

You can now de-assign me from this issue.

**Comment by wetneb** [comment]
Marking this as a good first issue, with the task being to put the upload button inside the contents of the "Preview" tab, not in the bar above, as was attempted before.

**Comment by smierz** [comment]
if I understand correctly, this issue is about placing a button in the preview tab of "Edit Wikibase schema" that calls the same function as directly selecting "Upload edits to Wikibase..." in the dropdown menu -> popup to login, etc

Suggestion:
I could place the button at the bottom (center) of the previews and would put the same text as in the corresponding menu entry "Upload edits to Wikibase..." for coherence. Optionally adding an "upload" icon at the beginning.
The menu entry will not be removed.

If that's what you are looking for, I'd like to be assigned :)

**Comment by wetneb** [comment]
I'm not sure about placing the button at the bottom, nor centering it horizontally. I don't think we center buttons horizontally anywhere so far, so to me that would be a bit inconsistent.

Placing it at the bottom would mean that users would need to scroll past all of the previewed items. Is this intentional, to make sure they visually scan all of the previewed entities before making the edits? To me it would be more usable and discoverable if the button could be at the top (say, in the top right corner of the tab contents).

**Comment by smierz** [comment]
Yes, the idea was to review the entries first before uploading. 

I will place it above the previews and see how it looks (as the text on the button is quite long + length depends on translation). An alternative might be to only place an "upload" icon on the button and show the text when hovering

**Comment by wetneb** [comment]
OK, I guess the bottom can also make sense and is likely easier to implement too.

**Comment by smierz** [comment]
Sounds good. I'll work on this then. So you can assign me the issue.

**Comment by smierz** [comment]
Here are some options to pick and choose from:

* button (text only)  below preview all the way to the right (as in schema-alignment-tab)
![Image](https://github.com/user-attachments/assets/60354ed6-a388-411d-9e61-b59652e68202)

* button (text only) above preview on the right
![Image](https://github.com/user-attachments/assets/e9960c70-eb04-4333-9306-5a3fd7824244)

* button (icon + text) above preview on the right, text on the left gets a line break
(alternatively the text could also be shortened to "Upload ...")
![Image](https://github.com/user-attachments/assets/15a24b21-294c-4b40-8f8b-bf439fb1654c)

* button (icon only) above preview on the right, text on the left gets a line break
![Image](https://github.com/user-attachments/assets/9927e625-b752-4e4a-a5f7-958104803663)

**Comment by magdmartin** [comment]
Thank you for the mock-up; it helps a lot to provide feedback. 

I'll go with the second option because: 
* The button is at the top, making it prominent and not at the end of a list (removing the need to scroll)
* As far as I know, nowhere in the application do we have an icon in a button (but please prove me wrong). I will avoid option 3 for that reason.
